toqito/matrix_ops/vector_to_density_matrix.py

```diff
@@ -0,0 +1,48 @@
+"""Convert row or column vector to density matrix."""
+import numpy as np
+
+
+def vector_to_density_matrix(input_array: np.ndarray) -> np.ndarray:
+    """Convert a given vector to a density matrix or return the density matrix if already given.
+
+    If the input is a vector, this function computes the outer product to form a density matrix.
+    If the input is already a density matrix (square matrix), it returns the matrix as is.
+
+    Examples
+    ==========
+
+    As an example, consider one of the Bell states.
+
+    >>> from toqito.states import bell
+    >>> from toqito.matrix_ops import vector_to_density_matrix
+    >>>
+    >>> vector_to_density_matrix(bell(0))
+    array([[0.5 0.  0.  0.5]
+    ...[0.  0.  0.  0. ]
+    ...[0.  0.  0.  0. ]
+    ...[0.5 0.  0.  0.5]])
```

Comment:
```suggestion
    array([[0.5 0.  0.  0.5]
               [0.  0.  0.  0. ]
               [0.  0.  0.  0. ]
               [0.5 0.  0.  0.5]])
```
I don't think you have to add `...` in the output. If you were defining the array in a code environment, then yes, you would have to use these.

i tried to get the formatting right with the output array. We can change this later after #459 is merged.